src/System.Private.ServiceModel/src/System/ServiceModel/Channels/TransportSecurityHelpers.cs

```diff
@@ -5,32 +5,199 @@
+                }
+                SspiSecurityTokenProvider tokenProvider = tokenManager.CreateSecurityTokenProvider(sspiRequirement) as SspiSecurityTokenProvider;
+                return tokenProvider;
+            }
+            return null;
+        }
+
+        public static SecurityTokenProvider GetDigestTokenProvider(
+            SecurityTokenManager tokenManager, EndpointAddress target, Uri via,
+            string transportScheme, AuthenticationSchemes authenticationScheme, ChannelParameterCollection channelParameters)
+        {
+            if (tokenManager != null)
+            {
+                InitiatorServiceModelSecurityTokenRequirement digestTokenRequirement =
+                    new InitiatorServiceModelSecurityTokenRequirement();
+                digestTokenRequirement.TokenType = ServiceModelSecurityTokenTypes.SspiCredential;
+                digestTokenRequirement.TargetAddress = target;
+                digestTokenRequirement.Via = via;
+                digestTokenRequirement.RequireCryptographicToken = false;
+                digestTokenRequirement.TransportScheme = transportScheme;
+                digestTokenRequirement.Properties[ServiceModelSecurityTokenRequirement.HttpAuthenticationSchemeProperty] = authenticationScheme;
+                if (channelParameters != null)
+                {
+                    digestTokenRequirement.Properties[ServiceModelSecurityTokenRequirement.ChannelParametersCollectionProperty] = channelParameters;
+                }
+                return tokenManager.CreateSecurityTokenProvider(digestTokenRequirement) as SspiSecurityTokenProvider;
+            }
+            return null;
+        }
+
+        public static SecurityTokenProvider GetCertificateTokenProvider(
+            SecurityTokenManager tokenManager, EndpointAddress target, Uri via, string transportScheme, ChannelParameterCollection channelParameters)
+        {
+            if (tokenManager != null)
+            {
+                InitiatorServiceModelSecurityTokenRequirement certificateTokenRequirement =
+                    new InitiatorServiceModelSecurityTokenRequirement();
+                certificateTokenRequirement.TokenType = SecurityTokenTypes.X509Certificate;
+                certificateTokenRequirement.TargetAddress = target;
+                certificateTokenRequirement.Via = via;
+                certificateTokenRequirement.RequireCryptographicToken = false;
+                certificateTokenRequirement.TransportScheme = transportScheme;
+                if (channelParameters != null)
+                {
+                    certificateTokenRequirement.Properties[ServiceModelSecurityTokenRequirement.ChannelParametersCollectionProperty] = channelParameters;
+                }
+                return tokenManager.CreateSecurityTokenProvider(certificateTokenRequirement);
+            }
+            return null;
+        }
```

Comment:
Don't we need to port some of the ones that create the `RecipientServiceModelSecurityTokenRequirement` as well?